### DISCUSSION TITLE
cache size

### DISCUSSION THREAD

**Discussion by Mr-Beast77** [cirruslabs/cirrus-ci-docs#815] [Ideas]
Hi dear,
If i am correct, default cache size is around 2G.  You can add an options to use users own google drive as cache storage, then it can be increased up to minimum 15G. If someone needs more than this 2G limit. I can manually do  this. But it would be nice if cirrus-ci would implement this.  Just a suggestion. 

Thanks for your awesome service. <3

**Comment by RDIL** [comment]
What are you caching that is that big?

**Reply by Mr-Beast77** [reply]
it is aosp project, from my testing it needed around 5.5G, though it hitted 1/3 files from ccache. 
Total hit/files in ccache = 55474/168553

I wish i could optimize it to ccache only those 55474 files. Any help would be appreciable. 
Thanks.

**Comment by fkorotkov** [comment]
There is no limit if you [bring you own compute service](https://cirrus-ci.org/guide/supported-computing-services/). For the managed instances the limit is `2G` indeed. We might increase it at some point but don't think it will be almost 10X.

Transferring such amounts of data and **unarchiving** is very costly and seems non-optimal. I'm not sure about what exactly are you trying to cache but if it's some kind caching artifacts for building your project then I'd investigate if you build system of choice can use  an [HTTP cache](https://cirrus-ci.org/guide/writing-tasks/#http-cache) for more granular caching.

**Reply by Mr-Beast77** [reply]
Thanks dear. 
Actually I got frightened by the documentation, since Gradle, Bazel and Pants are new to me. I will try to optimize the files of my ccache, and also learn the http cache. <3
